pkg/edgeview/src/edge-view.go

```diff
@@ -0,0 +1,580 @@
+	if *pdevip != "" && !runOnServer {
+		directQuery = true
+	}
+
+	var evStatus  types.EdgeviewStatus
+	pathStr := "/edge-view"
+	if directQuery { // ssh-mode
+		var err error
+		sshprivkey, err = ioutil.ReadFile("/ssh-private-key")
+		if err != nil {
+			fmt.Printf("ssh key file need to be docker mounted in '/ssh-private-key', error: %v\n", err)
+			return
+		}
+	} else {
+		// if wss endpoint is not passed in, try to get it from the JWT
+		if *ptoken != "" && *wsAddr == "" {
+			addrport, path, err := getAddrFromJWT(*ptoken, *pServer, &evStatus, *pInst)
+			if err != nil {
+				fmt.Printf("%v\n", err)
+				return
+			}
+			if path != "" {
+				pathStr = path
+			}
+			*wsAddr = addrport
+		}
+		if *wsAddr == "" {
+			fmt.Printf("wss address:port needs to be specified when '-token' is used\n")
+			return
+		}
+	}
+
+	initOpts()
+
+	var intSignal  chan os.Signal
+	var fstatus fileCopyStatus
+	remotePorts := make(map[int]int)
+	var tcpclientCnt int
+	var pqueryopt, pnetopt, psysopt, ppubsubopt, logopt, timeopt string
+	var jsonopt bool
+	typeopt := "all"
+	extraopt := 0
+	values := flag.Args()
+	var skiptype string
+	// the reason for this loop to get our own params is that it allows
+	// some options do not have to specify the "-something" in the front.
+	// the flag does not allow this.
+	// for example, put all the common usage in a script:
+	// ./myscript.sh log/<pattern>
+	// or ./myscript.sh log/<pattern> -time 0.2-0.5 -json
+	// or ./myscript.sh -device <ip-addr> route
+	for _, word := range values {
+		if skiptype != "" {
+			switch skiptype  {
+			case "dev":
+				*pdevip = word
+			case "time":
+				timeopt = word
+			case "type":
+				typeopt = word
+			case "extra":
+				numline, _ := strconv.Atoi(word)
+				extraopt = numline
+			case "token":
+				*ptoken = word
+			case "inst":
+			default:
+			}
+			skiptype = ""
+			continue
+		}
+		if strings.HasSuffix(word, "-help") || strings.HasSuffix(word, "-h") {
+			*phelpopt = true
+		} else if strings.HasSuffix(word, "-server") {
+			*pServer = true
+		} else if strings.HasSuffix(word, "-json") {
+			jsonopt = true
+		} else if strings.HasSuffix(word, "-debug") {
+			*pDebug = true
+		} else if strings.HasSuffix(word, "-device") {
+			skiptype = "dev"
+		} else if strings.HasSuffix(word, "-time") {
+			skiptype = "time"
+		} else if strings.HasSuffix(word, "-type") {
+			skiptype = "type"
+		} else if strings.HasSuffix(word, "-extra") {
+			skiptype = "extra"
+		} else if strings.HasSuffix(word, "-token") {
+			skiptype = "token"
+		} else if strings.HasSuffix(word, "-inst") {
+			skiptype = "inst"
+		} else {
+			pqueryopt = word
+		}
+	}
+
+	if *phopt || *phelpopt {
+		printHelp(pqueryopt)
+		return
+	}
+
+	if *pdevip == "" && *ptoken == "" {
+		fmt.Printf("either -device or -token option is needed\n")
+		return
+	}
+
+	// query option syntax checks
+	if pqueryopt != "" {
+		if strings.HasPrefix(pqueryopt, "log/") {
+			logs := strings.SplitN(pqueryopt, "log/", 2)
+			logopt = logs[1]
```

Comment:
Here logopt can be an arbitrary string. Where does it get used? Can we constrain it to be a valid filename syntax?

Reply:
the syntax 'log/<search-string>', is a string to narrow down the output in the log search. I'll put regex() for qualified string check.